src/connection_fallback.ts

```diff
@@ -0,0 +1,205 @@
+    }
+
+    this.state = state;
+  }
+
+  /** @private */
+  _onlineStatusChanged = (event: { type: string }) => {
+    this._log(`_onlineStatusChanged() - ${event.type}`);
+
+    if (event.type === 'offline') {
+      this._setState(ConnectionState.Closed);
+      this.cancelToken?.cancel('disconnect() is called');
+      this.cancelToken = undefined;
+      return;
+    }
+
+    if (event.type === 'online' && this.state === ConnectionState.Closed) {
+      this.connect(true);
+    }
+  };
+
+  /** @private */
+  _req = async <T = UR>(params: UnknownType, config: AxiosRequestConfig, retry: boolean): Promise<T> => {
+    if (!this.cancelToken && !params.close) {
+      this.cancelToken = axios.CancelToken.source();
+    }
+
+    try {
+      const res = await this.client.doAxiosRequest<T>(
+        'get',
+        (this.client.baseURL as string).replace(':3030', ':8900') + '/longpoll', // replace port if present for testing with local API
+        undefined,
+        {
+          config: { ...config, cancelToken: this.cancelToken?.token },
+          params,
+        },
+      );
+
+      this.consecutiveFailures = 0; // always reset in case of no error
+      return res;
+    } catch (err) {
+      this.consecutiveFailures += 1;
+
+      if (retry && isErrorRetryable(err)) {
+        this._log(`_req() - Retryable error, retrying request`);
+        await sleep(retryInterval(this.consecutiveFailures));
+        return this._req<T>(params, config, retry);
+      }
+
+      throw err;
+    }
+  };
+
+  /** @private */
+  _poll = async () => {
+    while (this.state === ConnectionState.Connected) {
+      try {
+        const data = await this._req<{
+          events: Event<AttachmentType, ChannelType, CommandType, EventType, MessageType, ReactionType, UserType>[];
+        }>({}, { timeout: 30000 }, true); // 30s => API responds in 20s if there is no event
+
+        if (data.events?.length) {
+          for (let i = 0; i < data.events.length; i++) {
+            this.client.dispatchEvent(data.events[i]);
+          }
+        }
+      } catch (err) {
+        if (axios.isCancel(err)) {
+          this._log(`_poll() - axios canceled request`);
+          return;
+        }
+
+        /** client.doAxiosRequest will take care of TOKEN_EXPIRED error */
+
+        if (isConnectionIDError(err)) {
+          this._log(`_poll() - ConnectionID error, connecting without ID...`);
+          this._setState(ConnectionState.Disconnected);
+          this.connect(true);
+          return;
+        }
+
+        if (isAPIError(err) && !isErrorRetryable(err)) {
+          this._setState(ConnectionState.Closed);
+          return;
+        }
+
+        await sleep(retryInterval(this.consecutiveFailures));
+      }
+    }
+  };
+
+  /**
+   * connect try to open a longpoll request
+   * @param reconnect should be false for first call and true for subsequent calls to keep the connection alive and call recoverState
+   */
+  connect = async (reconnect = false) => {
```

Comment:
this connect has `reconnect` as argument, while the other has `timeout`. Maybe better to use an object on both.
```suggestion
  connect = async (options = { reconnect: false }) => {
```